selfdrive/ui/android_ui.cc

```diff
@@ -243,109 +226,40 @@ int main(int argc, char* argv[]) {
-    }
-
     // Don't waste resources on drawing in case screen is off
-    if (s->awake) {
-      ui_draw(s);
-      glFinish();
-      should_swap = true;
+    if (!s->awake) {
+      continue;
     }
 
-    s->sound.setVolume(fmin(MAX_VOLUME, MIN_VOLUME + s->scene.controls_state.getVEgo() / 5)); // up one notch every 5 m/s
-
-    bool controls_timeout = (s->sm->frame - s->sm->rcv_frame("controlsState")) > 5*UI_FREQ;
-    if (s->started && !s->scene.frontview && controls_timeout) {
-      if (!s->controls_seen) {
-        // car is started, but controlsState hasn't been seen at all
-        s->scene.alert_text1 = "openpilot Unavailable";
-        s->scene.alert_text2 = "Waiting for controls to start";
-        s->scene.alert_size = cereal::ControlsState::AlertSize::MID;
-      } else {
-        // car is started, but controls is lagging or died
-        LOGE("Controls unresponsive");
-
-        if (s->scene.alert_text2 != "Controls Unresponsive") {
-          s->sound.play(AudibleAlert::CHIME_WARNING_REPEAT);
-        }
-
-        s->scene.alert_text1 = "TAKE CONTROL IMMEDIATELY";
-        s->scene.alert_text2 = "Controls Unresponsive";
-        s->scene.alert_size = cereal::ControlsState::AlertSize::FULL;
-        update_status(s, STATUS_ALERT);
-      }
-      ui_draw_vision_alert(s, s->scene.alert_size, s->status, s->scene.alert_text1.c_str(), s->scene.alert_text2.c_str());
-    }
+    // up one notch every 5 m/s
+    s->sound.setVolume(fmin(MAX_VOLUME, MIN_VOLUME + s->scene.controls_state.getVEgo() / 5));
 
+    // set brightness
+    float clipped_brightness = fmin(512, (s->light_sensor*brightness_m) + brightness_b);
+    smooth_brightness = fmin(255, clipped_brightness * 0.01 + smooth_brightness * 0.99);
+    ui_set_brightness(s, (int)smooth_brightness);
 
-    if (s->sm->frame % (2*UI_FREQ) == 0) {
-      read_param(&s->is_metric, "IsMetric");
-    } else if (s->sm->frame % (3*UI_FREQ) == 0) {
-      int param_read = read_param(&s->last_athena_ping, "LastAthenaPingTime");
-      if (param_read != 0) { // Failed to read param
-        s->scene.athenaStatus = NET_DISCONNECTED;
-      } else if (nanos_since_boot() - s->last_athena_ping < 70e9) {
-        s->scene.athenaStatus = NET_CONNECTED;
-      } else {
-        s->scene.athenaStatus = NET_ERROR;
-      }
-    }
     update_offroad_layout_state(s, pm);
 
-    pthread_mutex_unlock(&s->lock);
-
-    // the bg thread needs to be scheduled, so the main thread needs time without the lock
-    // safe to do this outside the lock?
-    if (should_swap) {
-      double u2 = millis_since_boot();
-      if (u2-u1 > 66) {
-        // warn on sub 15fps
-        LOGW("slow frame(%d) time: %.2f", draws, u2-u1);
-      }
-      draws++;
-      framebuffer_swap(s->fb);
+    ui_draw(s);
+    double u2 = millis_since_boot();
+    if (!s->scene.frontview && (u2-u1 > 66)) {
+      // warn on sub 15fps
+      LOGW("slow frame(%d) time: %.2f", (s->sm)->frame, u2-u1);
```

Comment:
(%d) should be (%llu)

Reply:
s->sm->frame may be better?

Reply:
what's the difference with the second thing?